<launch>
  <arg name="output_screen" default="false"/>
  <arg name="image_topic" default="/rs_d435/color/image_rect_color" />
  <arg name="camera_info_topic" default="/rs_d435/color/camera_info" />
  <arg name="debug_view" default="true"/>
  <arg name="size_filter" default="true"/>
  <arg name="object_height" default="0.2"/>
  <arg name="contour_area_size" default="0.06"/>
  <arg name="contour_area_margin" default="0.02"/>
  <arg name="contour_area_min_thre" default="0.01"/>
  <arg name="frame_id" default="target_object_color"/>
  <arg name="h_limit_max" default="0" doc="The maximum allowed field value Hue" />
  <arg name="h_limit_min" default="316" doc="The minimum allowed field value Hue" />
  <arg name="l_limit_max" default="255" doc="The maximum allowed field value Saturation" />
  <arg name="l_limit_min" default="50" doc="The minimum allowed field value Saturation" />
  <arg name="s_limit_max" default="255" doc="The maximum allowed field value Value" />
  <arg name="s_limit_min" default="76" doc="The minimum allowed field value Value" />

  <node pkg="nodelet" type="nodelet" name="object_detection_nodelet" args="manager" output="screen" if="$(arg output_screen)"/>
  <node pkg="nodelet" type="nodelet" name="object_detection_nodelet" args="manager" unless="$(arg output_screen)"/>

  <node pkg="nodelet" type="nodelet" name="hls_color_filter"
        args="load opencv_apps/hls_color_filter object_detection_nodelet">
    <remap from="image" to="$(arg image_topic)"/>
    <param name="debug_view" value="false"/>
    <param name="h_limit_max" value="$(arg h_limit_max)"/>
    <param name="h_limit_min" value="$(arg h_limit_min)"/>
    <param name="l_limit_max" value="$(arg l_limit_max)"/>
    <param name="l_limit_min" value="$(arg l_limit_min)"/>
    <param name="s_limit_max" value="$(arg s_limit_max)"/>
    <param name="s_limit_min" value="$(arg s_limit_min)"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="ground_object_detection_with_size_filter"
        args="load aerial_robot_perception/GroundObjectDetection object_detection_nodelet" output="screen">
    <remap from="image" to="/hls_color_filter/image" />
    <remap from="camera_info" to="$(arg camera_info_topic)"/>
    <param name="object_height" value="$(arg object_height)"/>
    <param name="contour_area_size" value="$(arg contour_area_size)"/>
    <param name="contour_area_margin" value="$(arg contour_area_margin)"/>
    <param name="contour_area_min_thre" value="$(arg contour_area_min_thre)"/>
    <param name="debug_view" value="$(arg debug_view)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
  </node>

</launch>
